core/unix/signal.c

```diff
@@ -1036,6 +1120,12 @@ signal_thread_inherit(dcontext_t *dcontext, void *clone_record)
             info->sighand->is_shared = true;
             d_r_mutex_lock(&info->sighand->lock);
             info->sighand->refcount++;
+            for (i = 1; i <= MAX_SIGNUM; i++) {
+                if (!EMULATE_SIGMASK(info, i))
+                    continue;
+                if (!kernel_sigismember(&info->app_sigblocked, i))
+                    ATOMIC_INC(int, info->sighand->threads_unmasked[i]);
```

Comment:
Does this mean we keep a common count for the cloned child and the parent? But the parent and child are different processes, so signals shouldn't be rerouted between them right?

Reply:
This code is for two entities that share a signal handler table, created by SYS_clone with CLONE_SIGHAND.  By definition this is the group that will share signals and should have signals rerouted between them.  It used to be you could create weird mixtures of sharing with separate address spaces but identical handlers but they've locked that down in recent kernels, so nowadays this is always two "threads".